tap_square/streams.py

```diff
@@ -98,9 +102,10 @@ class Payments():
     object_type = 'DISCOUNT'
 
     def sync(self, client, start_time, bookmarked_cursor): #pylint: disable=no-self-use
-        for page, cursor in client.get_payments(client, start_time, bookmarked_cursor):
+        for page, cursor in client.get_payments(self.object_type, start_time, bookmarked_cursor):
```

Comment:
It is unclear to me what this does. The other stream classes pass in client here.

Reply:
See my comment below which explains it and shows the error